File: src/constants/composerSpeech.ts
/** Tauri 事件：composer 本地/流式语音转写 partial 或 final 结果。 */
export const COMPOSER_SPEECH_TRANSCRIPT_EVENT = "composer-speech-transcript" as const;

/** 运行时实际使用的听写引擎。 */
export type ComposerSpeechEngine = "sensevoice" | "web";

export interface ComposerSpeechTranscriptPayload {
  sessionId: string;
  transcript: string;
  isFinal: boolean;
  /** 识别失败时的本地化说明（final 且无 transcript 时常见）。 */
  error?: string;
}

File: src-tauri/src/composer_speech.rs
//! Composer 语音听写：前后端共享事件名与 payload。

pub const COMPOSER_SPEECH_TRANSCRIPT_EVENT: &str = "composer-speech-transcript";

#[derive(Clone, serde::Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ComposerSpeechTranscriptPayload {
    pub session_id: String,
    pub transcript: String,
    pub is_final: bool,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub error: Option<String>,
}
